src/rez/utils/data_utils.py

```diff
@@ -213,50 +214,53 @@ def get_dict_diff_str(d1, d2, title):
     return '\n'.join(lines)
 
 
-class cached_property(object):
-    """Simple property caching descriptor.
-
-    Example:
-
-        >>> class Foo(object):
-        >>>     @cached_property
-        >>>     def bah(self):
-        >>>         print('bah')
-        >>>         return 1
-        >>>
-        >>> f = Foo()
-        >>> f.bah
-        bah
-        1
-        >>> f.bah
-        1
-    """
-    def __init__(self, func, name=None):
-        self.func = func
-        # Make sure that Sphinx autodoc can follow and get the docstring from our wrapped function.
-        functools.update_wrapper(self, func)
-        self.name = name or func.__name__
-
-    def __get__(self, instance, owner=None):
-        if instance is None:
-            return self
-
-        result = self.func(instance)
-        try:
-            setattr(instance, self.name, result)
-        except AttributeError:
-            raise AttributeError("can't set attribute %r on %r"
-                                 % (self.name, instance))
-        return result
-
-    # This is to silence Sphinx that complains that cached_property is not a callable.
-    def __call__(self):
-        raise RuntimeError("@cached_property should not be called.")
-
-    @classmethod
-    def uncache(cls, instance, name):
-        if hasattr(instance, name):
-            delattr(instance, name)
+if TYPE_CHECKING:
+    cached_property = property
+else:
+    class cached_property(object):
```

Comment:
it's much easier to pretend that `cached_property` is `property` than to type hint all the subtleties of a descriptor.

Reply:
But we loose the `uncache` method.

Reply:
`typing.TYPE_CHECKING` always resolve to `False` at runtime and `True` only during static analysis.  So the code within the `if TYPE_CHECKING` block will never run.  It's a way to simplify certain type analysis situations that arise.

Reply:
>typing.TYPE_CHECKING always resolve to False

I know, but we are loosing stuff during typing. That's my whole point (the same apply to all my comments that are similar to this one).

Reply:
in what way would the static analysis be degraded by using `property` instead of `cached_property`?  To my knowledge, they are functionally equivalent from a static analysis POV:  the types returned are the same.